### PR TITLE
replace username with name and surname

### PR DESCRIPTION
**Pull request by mr-loop-1**
Replaces the username field with name and surname in the application

### CODE REVIEW

**Comment by mr-loop-1** [comment]
https://github.com/bluewave-labs/bluewave-onboarding/pull/181#discussion_r1744603745

@erenfn I changed username to name and surname in backend too and hence on frontend where it is used.

**Comment by gorkem-bwl** [comment]
Thank you for your contribution @mr-loop-1

**Comment by mr-loop-1** [comment]
@erenfn now I saw your pr but it was closed

**Review by uparkalau — CHANGES_REQUESTED** [review]
do not delete username from the table

**Comment by erenfn** [comment]
> @erenfn now I saw your pr but it was closed

It is still open, but you may need to merge it since you made a change afterwards

**Comment by mr-loop-1** [comment]
@uparkalau @erenfn do we need username field too with surname and name

**Comment by uparkalau** [comment]
> > do not delete username from the table
> 
> @uparkalau @erenfn do we need username field too with surname and name

Yes keep username do not delete it

**Comment by mr-loop-1** [comment]
> Yes keep username do not delete it

then I guess it would be a unique identifier like email. Btw what would be displayed on Header/dashboard and stored in cookies, the name+surname or username

File: backend/migrations/20240422214219-create_users_table.js [hunk 1]
```diff
@@ -9,10 +9,12 @@ module.exports = {
         primaryKey: true,
         autoIncrement: true,
       },
-      username: {
+      name: {
```
Comment (uparkalau) [inline backend/migrations/20240422214219-create_users_table.js:12]: This is not the correct way to add new fields to the database. You don't need to change the existing migration; instead, you should create a new migration with the new fields you want to add and remove the field from the table. So, create a new migration. Yes, you can delete the username field. On the frontend, create a function that generates the username from the name and surname, and use this function wherever a username is needed.

**Review by uparkalau — CHANGES_REQUESTED** [review]

File: backend/migrations/20240422214219-create_users_table.js (hunk 1 above)
Comment (mr-loop-1) [reply backend/migrations/20240422214219-create_users_table.js]: Yes, that was an oversight. It was when I wanted to quickly test with name and surname fields I modified the existing, but didn't create new one when pushing.

**Review by uparkalau — APPROVED** [review]

**Review by erenfn — APPROVED** [review]

**Pull request merged**